src/datasources/asset/components/editors/asset-summary/AssetSummaryDataSource.ts

```diff
@@ -17,10 +19,35 @@ export class AssetSummaryDataSource extends AssetDataSourceBase {
     defaultQuery = {
     };
 
-    runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-        throw new Error("Method not implemented.");
+    async runQuery(query: AssetQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
+        return this.processMetadataQuery(query as AssetQuery);
+    }
+
+    shouldRunQuery(_: AssetQuery): boolean {
+        return true;
     }
-    shouldRunQuery(query: AssetQuery): boolean {
-        throw new Error("Method not implemented.");
+
+    async processMetadataQuery(query: AssetQuery) {
+        const result: DataFrameDTO = { refId: query.refId, fields: [] };
+
+        const assets: AssetSummaryQuery = await this.getAssetSummary();
+
+        result.fields = [
+            { name: 'Total', values: [assets.total] },
```

Comment:
Since we are reusing these values in the tests, we might want to create a constants for this

Reply:
I'm not sure if we should change it now since we have this just here once and once in the test. If we'll add more, we might consider adding a constant

Reply:
Added constant